diff --git a/clang/lib/Format/ContinuationIndenter.cpp b/clang/lib/Format/ContinuationIndenter.cpp
index 0ca297a5f..2dba2e8dd 100644
--- a/clang/lib/Format/ContinuationIndenter.cpp
+++ b/clang/lib/Format/ContinuationIndenter.cpp
@@ -367,7 +367,7 @@ bool ContinuationIndenter::mustBreak(const LineState &State) {
         Current.isBlockIndentedInitRBrace(Style)))) {
     return true;
   }
-  if (CurrentState.BreakBeforeClosingParen && Current.is(tok::r_paren))
+  if (CurrentState.BreakBeforeClosingParen && Current.isOneOf(tok::r_paren, TT_TemplateCloser))
     return true;
   if (Style.Language == FormatStyle::LK_ObjC &&
       Style.ObjCBreakBeforeNestedBlockParam &&
@@ -1050,7 +1050,7 @@ unsigned ContinuationIndenter::addTokenOnNewLine(LineState &State,
     CurrentState.BreakBeforeClosingBrace = true;
   }

-  if (PreviousNonComment && PreviousNonComment->is(tok::l_paren)) {
+  if (PreviousNonComment && PreviousNonComment->isOneOf(tok::l_paren, TT_TemplateOpener)) {
     CurrentState.BreakBeforeClosingParen =
         Style.AlignAfterOpenBracket == FormatStyle::BAS_BlockIndent;
   }
@@ -1170,7 +1170,7 @@ unsigned ContinuationIndenter::getNewLineColumn(const LineState &State) {
     return State.Stack[State.Stack.size() - 2].LastSpace;
   }
   if (Style.AlignAfterOpenBracket == FormatStyle::BAS_BlockIndent &&
-      (Current.is(tok::r_paren) ||
+      (Current.isOneOf(tok::r_paren, TT_TemplateCloser) ||
        (Current.is(tok::r_brace) &&
         Current.MatchingParen->is(BK_BracedInit))) &&
       State.Stack.size() > 1) {
diff --git a/clang/lib/Format/TokenAnnotator.cpp b/clang/lib/Format/TokenAnnotator.cpp
index 73840332e..a7317fba2 100644
--- a/clang/lib/Format/TokenAnnotator.cpp
+++ b/clang/lib/Format/TokenAnnotator.cpp
@@ -5482,8 +5482,6 @@ bool TokenAnnotator::canBreakBefore(const AnnotatedLine &Line,
   if (Right.is(TT_ImplicitStringLiteral))
     return false;

-  if (Right.is(TT_TemplateCloser))
-    return false;
   if (Right.is(tok::r_square) && Right.MatchingParen &&
       Right.MatchingParen->is(TT_LambdaLSquare)) {
     return false;
@@ -5491,7 +5489,7 @@ bool TokenAnnotator::canBreakBefore(const AnnotatedLine &Line,

   // We only break before r_brace if there was a corresponding break before
   // the l_brace, which is tracked by BreakBeforeClosingBrace.
-  if (Right.is(tok::r_brace)) {
+  if (Right.isOneOf(tok::r_paren, TT_TemplateCloser)) {
     return Right.MatchingParen && (Right.MatchingParen->is(BK_Block) ||
                                    (Right.isBlockIndentedInitRBrace(Style)));
   }
